airflow-core/src/airflow/jobs/asyncio_monitor.py

```diff
@@ -0,0 +1,244 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import dataclasses
+import logging
+import sys
+import threading
+import time
+import traceback
+from collections import deque
+from time import perf_counter
+from types import FrameType
+from typing import TYPE_CHECKING
+
+import structlog
+
+from airflow._shared.timezones import timezone
+from airflow.stats import Stats
+
+if TYPE_CHECKING:
+    from structlog.typing import FilteringBoundLogger
+
+logger = logging.getLogger(__name__)
+
+
+log: FilteringBoundLogger = structlog.get_logger(logger_name=__name__)
+
+
+@dataclasses.dataclass
+class StallSample:
+    """Single stack capture during a stall."""
+
+    taken_at_perf: float
+    stack_text: str
+
+
+@dataclasses.dataclass
+class StallIncident:
+    """A coalesced stall incident with start/end, samples, and suspected tasks."""
+
+    started_at_perf: float
+    ended_at_perf: float | None
+    threshold: float
+    samples: list[StallSample]
+    suspected_tasks: list[str]  # filled on correlation
```

Comment:
i think `suspected_tasks` is no longer used. i would review this interface to make sure there's nothing there that isn't used